src/main_thread/api/metricsCollector.ts

```diff
@@ -0,0 +1,70 @@
+import type { SegmentSinkMetrics } from "../../core/segment_sinks/segment_buffers_store";
```

Comment:
I find the inclusion of a whole file at the API level then communicated to Init for those optional debug information a little too heavy.

We could for example just, instead of providiing a `SegmentSinksStore` instance on `loaded` have a function returning the content of the buffers - which may be wrapped into either a promise system or some `SharedReference` no?